__tests__/integration.js

```diff
@@ -0,0 +1,391 @@
+const awsServerlessExpress = require('../index')
+const apiGatewayEvent = require('../example/api-gateway-event.json')
+const app = require('../example/app')
+
+const server = awsServerlessExpress.createServer(app)
+const lambdaFunction = {
+  handler: (event, context) => awsServerlessExpress.proxy(server, event, context)
+}
+
+function clone(json) {
+  return JSON.parse(JSON.stringify(json))
+}
+
+function makeEvent(eventOverrides) {
+  const baseEvent = clone(apiGatewayEvent)
+  const headers = Object.assign({}, baseEvent.headers, eventOverrides.headers)
+  delete eventOverrides.headers
+  const root = Object.assign({}, baseEvent, eventOverrides)
+  root.headers = headers
+  return root
+}
+
+function expectedRootResponse() {
+  return makeResponse({
+    "headers": {
+      "content-length": "3747",
+      "content-type": "text/html; charset=utf-8",
+      "etag": "W/\"ea3-WawLnWdlaCO/ODv9DBVcX0ZTchw\""
+    }
+  })
+}
+
+function makeResponse(response) {
+  const baseResponse = {
+    "body": "",
+    "isBase64Encoded": false,
+    "statusCode": 200
+  }
+  const baseHeaders = {
+    "access-control-allow-origin": "*",
+    "connection": "close",
+    "content-type": "application/json; charset=utf-8",
+    "x-powered-by": "Express"
+  }
+  const headers = Object.assign({}, baseHeaders, response.headers)
+  const finalResponse = Object.assign({}, baseResponse, response)
+  finalResponse.headers = headers
+  return finalResponse
+}
+
+describe('integration tests', () => {
+  test('GET HTML (initial request)', (done) => {
+    const succeed = response => {
```

Comment:
The original resolution for Node.js Lambda functions is `context.succeed(response)` which this library calls. You can see on line 62/63 (and all other tests) that we specify our `context` object as the second parameter of `makeEvent`. We make all our assertions against the `response` of a `context.succeed` call